drivers/hmis_external_apis/app/jobs/hmis_external_apis/ac_hmis/fetch_clients_with_active_referrals_job.rb

```diff
@@ -0,0 +1,45 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# An HMIS User indicates that there is a vacancy in a program.
+module HmisExternalApis::AcHmis
+  class FetchClientsWithActiveReferralsJob < ApplicationJob
+    include HmisExternalApis::AcHmis::ReferralJobMixin
+
+    def perform
+      mci_ids = link.active_referral_mci_ids.parsed_body
+      client_ids = HmisExternalApis::ExternalId.where(
+        namespace: HmisExternalApis::AcHmis::Mci::SYSTEM_ID,
+        value: mci_ids.map(&:to_s),
+        source_type: 'Hmis::Hud::Client',
+      ).pluck(:source_id)
+
+      cded = Hmis::Hud::CustomDataElementDefinition.where(key: :has_active_referrals_in_link).first_or_create!(
+        owner_type: 'Hmis::Hud::Client',
+        field_type: :boolean,
+        key: :has_active_referrals_in_link,
+        label: 'Has Active Referrals in LINK',
+        repeats: false,   # client can only have 1 value
+        data_source: data_source,
+        user: system_user,
+      )
+
+      Hmis::Hud::Client.
+        where(id: client_ids).
+        each do |client|
+          client.custom_data_elements.
+            where(data_element_definition: cded).
+            first_or_create!(value_boolean: true, data_source: data_source, user: system_user).
+            update!(value_boolean: true)
+        end
```

Comment:
This will make at least 1 query per client, I don't think that's necessary.
Can you rework this to remove the n+1?

Using `import` could be a good option. `on_duplicate_key_update` can be used to update the `value_boolean` and `user` if an existing record is found.